lib/runners/harness.rb

```diff
@@ -91,5 +94,33 @@ def handle_error(exn)
         trace_writer.error "#{exn.message} (#{exn.class})"
       end
     end
+
+    def remove_ignored_files(config)
+      ignores = config.ignore
+      return if ignores.empty?
+
+      trace_writer.message("Deleting specified files via the `ignore` option...") do
+        with_gitignore(ignores) do |list|
+          FileUtils.rm_rf(list)
+          trace_writer.message list.map { |f| f.relative_path_from(working_dir) }.join("\n")
+        end
+      end
+    end
+
+    def with_gitignore(ignores)
+      gitignore = (working_dir / ".gitignore")
+
+      backup = gitignore.file? ? gitignore.read : nil
+      gitignore.write(ignores.join("\n"))
+
+      shell = Shell.new(current_dir: working_dir, trace_writer: trace_writer, env_hash: {})
+      shell.capture3!("git", "init")
```

Comment:
[suggestion] I'm concerned that `git init` output may be unuseful and make confused users, so what about suppressing as follow?

```suggestion
      shell.capture3!("git", "init", trace_command_line: false, trace_stdout: false)
```

```diff
diff --git a/lib/runners/shell.rb b/lib/runners/shell.rb
index 47db1296..376446b7 100644
--- a/lib/runners/shell.rb
+++ b/lib/runners/shell.rb
@@ -92,13 +92,14 @@ module Runners
 
     def capture3_trace(command, *args, **options)
       # @type var options: any
+      trace_command_line = options.fetch(:trace_command_line, true)
       trace_stdout = options.fetch(:trace_stdout, true)
       trace_stderr = options.fetch(:trace_stderr, true)
       raise_on_failure = options.fetch(:raise_on_failure, false)
       is_success = options.fetch(:is_success) { ->(status) { status.success? } }
 
       command_line = [command] + args
-      trace_writer.command_line(command_line)
+      trace_writer.command_line(command_line) if trace_command_line
 
       Open3.capture3(env_hash, command, *args, { chdir: current_dir.to_s }).tap do |stdout_str, stderr_str, status|
         trace_writer.stdout stdout_str if trace_stdout
```

Reply:
Good. I updated with 9e49f19.